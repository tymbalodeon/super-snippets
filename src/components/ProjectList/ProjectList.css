.ProjectList__add-button {
  padding: 5px;
  margin-left: 5px;
  margin-right: 5px;
  background-color: #cbccc6;
  color: #1f2430;
  font-size: 12px;
  width: 120px;
  /* white-space: nowrap; */
}

.ProjectList {
  text-align: center;
}

select {
  margin-bottom: 20px;
}
