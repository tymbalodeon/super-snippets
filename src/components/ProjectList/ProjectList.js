import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import SnippetApiService from '../../services/snippet-api-service';
import './ProjectList.css';

export default class ProjectList extends Component {
  state = {
    project_id: this.props.project_id || ''
  };

  renderProjects() {
    const { projects, updateProjectId } = this.props;
    return (
      <div className="project-list-wrapper">
        <label htmlFor="snippet-project-select">Filter by Project: </label>
        <select
          id="snippet-project-select"
          name="project_id"
          onChange={e => updateProjectId(e.target.value)}
        >
          <option key="0" value={''}>
            --DISPLAY ALL--
          </option>
          {projects.map(project => (
            <option key={project.id} value={project.id}>
              {project.title}
            </option>
          ))}
        </select>
      </div>
    );
  }

  componentDidMount() {
    if (localStorage.getItem('super-snippets-client-auth-token'))
      SnippetApiService.getProjects()
        .then(res => this.props.setProjects(res))
        .catch(error => console.log({ error }));
  }

  render() {
    if (localStorage.getItem('super-snippets-client-auth-token')) {
      return (
        <div className="ProjectList">
          {this.renderProjects()}
          <div className="ProjectList__button-wrapper">
            <Link to="/add-project">
              <button className="ProjectList__add-button">
                <span className="add-button">CREATE PROJECT</span>
              </button>
            </Link>
            <Link to="/add-snippet">
              <button className="ProjectList__add-button">
                <span className="add-button">CREATE SNIPPET</span>
              </button>
            </Link>
          </div>
        </div>
      );
    } else {
      return '';
    }
  }
}
