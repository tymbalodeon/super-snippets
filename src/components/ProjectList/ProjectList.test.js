import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ProjectList from './ProjectList';

describe('<ProjectList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<ProjectList />, div);

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer.create(<ProjectList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
