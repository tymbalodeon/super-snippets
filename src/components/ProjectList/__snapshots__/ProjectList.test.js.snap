// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectList /> renders the UI as expected 1`] = `""`;
